ivy/functional/frontends/paddle/nn/functional/loss.py

```diff
@@ -287,11 +309,22 @@ def nll_loss(
         output = ivy.sum(loss)
         if ignore_index >= 0 and ignore_index < ivy.shape(input)[1]:
             output = output - loss[ignore_index]
-        return output
-    num = ivy.sum(loss)
-    output = num / den
-    if ignore_index >= 0 and ignore_index < ivy.shape(input)[1]:
-        output = output - loss[ignore_index] / den
+
+    elif reduction == "mean":
+        output = ivy.mean(loss)
+        if ignore_index >= 0 and ignore_index < ivy.shape(input)[1]:
+            output = output - loss[ignore_index]
+
+    elif reduction == "none":
+        output = loss
+        if ignore_index >= 0 and ignore_index < ivy.shape(input)[1]:
+            output = output - loss[ignore_index] / den
```

Comment:
Could you explain your reasoning here? This is not what I would have expected here